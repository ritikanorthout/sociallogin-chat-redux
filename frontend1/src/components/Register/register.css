body,
#root {
  height: 100%;
}
.login_container {
  background-color: #f4f5f7;
  height: 100vh;
}
.logn_bg {
  position: relative;
  background-color: #ffccff;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
}
.brand_log {
  width: 300px;
}

.reg-container {
  margin-top: 100px;
  margin-left: 400px;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: white;
  border-style: groove;
  border-color: #f4f5f7;
  height: 550px;
  width: 400px;
}
.reg-container h1 {
  text-align: center;
  text-transform: uppercase;
  font-size: 25px;
}

.reg_btn {
  background-color: cornflowerblue;
  border-radius: 10px;
  order: none;
  padding: 8px 30px;
  margin: 5px;
  width: 350px;
}


.error{
  color: red;
  margin-top: -15px;
}