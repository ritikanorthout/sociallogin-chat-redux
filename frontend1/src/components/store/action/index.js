import { AUTH_LOGIN, USER_REGISTER } from "../../constant/type";

// export const fngetLogin = (data) => {
//   return (dispatch) => {
//     dispatch({
//       type: AUTH_LOGIN,
//       payload: data,
//     });
//   };
// };

// export const fnuserRegister = (userData) => {
//   return (dispatch) => {
//     dispatch({
//       type: USER_REGISTER,
//       payload: userData,
//     });
//   };
// };

export const getLogin = (data) => {
  return (dispatch) => {
    dispatch({
      type: AUTH_LOGIN,
      payload: data,
    });
  };
};

export const userRegister = (userData) => {
  return (dispatch) => {
    dispatch({
      type: USER_REGISTER,
      payload: userData,
    });
  };
};
