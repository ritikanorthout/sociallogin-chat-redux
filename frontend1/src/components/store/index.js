import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import Reducer from "./reducer";

function saveToLocalStorage(state) {

  console.log('saveToLocalStorage ', state);

  try {
    const serializedState = JSON.stringify(state);
    localStorage.setItem("state", serializedState);
  } catch (error) {
    console.log(error);
  }
}

function loadToStore() {

  console.log('loadToStore');
  try {
    const serializedState = localStorage.getItem("state");
    console.log('serializedState',serializedState);
    if (serializedState === null) return undefined;
    return JSON.parse(serializedState);
  } catch (error) {
    console.log(error);
  }
}
const per = loadToStore();
const store = createStore(Reducer, per, applyMiddleware(thunk));

store.subscribe(() => saveToLocalStorage(store.getState()));
export default store;
