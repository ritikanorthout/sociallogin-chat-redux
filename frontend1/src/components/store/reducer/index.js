import { AUTH_LOGIN, USER_REGISTER, LOGOUT } from "../../constant/type";
const initialState = {
  isLoggedin: false,
  userInfo: { name: "", email: "", mobile: "", token: "", imgUrl: "" },
  userList: [],
};

const Reducer = (state = initialState, action) => {
  console.log("action.payload : ",action.payload);

  console.log("state : ",state);

  switch (action.type) {
    case AUTH_LOGIN:
      return {
        ...state,
        isLoggedin: true,
        userInfo: {
          name: action.payload.name,
          email: action.payload.email,
          imgUrl: action.payload.imageUrl,
        },
      };
    case USER_REGISTER:
      return {
        ...state,
        userList: [...state.userList, action.payload],
      };
    case LOGOUT:
      return {
        ...state,
        userInfo: { name: "", email: "", mobile: "", token: "", imgUrl: "" },
        isLoggedin: false,
      };
    default:
      return {
        ...state,
      };
  }

};

export default Reducer;
