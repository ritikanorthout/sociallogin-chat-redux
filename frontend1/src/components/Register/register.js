import React, { useState } from "react";
import { Col, Form, FormGroup, Label, Input, Button } from "reactstrap";
import { Link, Redirect } from "react-router-dom";
import axios from 'axios';
import "./register.css";
import { useDispatch } from "react-redux";
import * as AppConstants from '../../constants/AppConstants';
import { USER_REGISTER } from "../../components/constant/type";

function Register() {
  const dispatch = useDispatch();

  const fnRegister = (res) => {

    console.log('res',res);

    if(res != "" && res.email != "" &&  res.name != "" && res.mobile != "" && res.password != "")
    {
      dispatch({ type: USER_REGISTER, payload: res });
      alert("Registered successfully!");
      // const BaseURL = 'http://localhost/api/save_user.php';
      // axios.post(BaseURL, {
      //   res
      //   })
      //   .then((response) => {
      //     if(response)
      //     {
      //       console.log(response);
      //       dispatch({ type: USER_REGISTER, payload: res });
      //       alert("Registered successfully!");
      //     }
          
      //   }, (error) => {
      //     console.log(error);
      //   });   
    }
  };

  const [userInfo, setuserInfo] = useState({
    name: "",
    password: "",
    email: "",
    mobile: "",
  });

  const [errors, setErrors] = useState({
    name: "",
    password: "",
    email: "",
    mobile: "",
  });
  const [success, setsuccess] = useState(false);

  const fnonChange = (event) => {
     const { name, value } = event.target;
     let errors = {};
     let error = '';
     const validEmailRegex = 
     RegExp(/^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i);
     switch (name) {

       case 'name':
        if((/\s/.test(value)))
        {
         error = errors.name =  AppConstants.VALIDATION_INVALID_NAME;
        }
        else if(value.length > 50)
        {
         error =  errors.name = AppConstants.VALIDATION_NAME_MAX_CHAR;
        }
        else
        {
         error =  errors.name = ''; 
        }
        break;

        case 'email':
         if(!validEmailRegex.test(value))
         {
          error = errors.email =  AppConstants.VALIDATION_INVALID_EMAIL;
         }
         else if(value.length > 254)
         {
          error =  errors.email = AppConstants.VALIDATION_EMAIL_MAX_CHAR;
         }
         else
         {
          error =  errors.email = ''; 
         }
         break;

         case 'mobile':
          if((/\s/.test(value)) || !(/^-?[\d.]+(?:e-?\d+)?$/.test(value)))
          {
           error = errors.mobile =  AppConstants.VALIDATION_INVALID_MOBILE;
          }
          else if(value.length < 10)
          {
           error =  errors.mobile = AppConstants.VALIDATION_MOBILE_MIN_CHAR;
          }
          else if(value.length > 10)
          {
           error =  errors.mobile = AppConstants.VALIDATION_MOBILE_MAX_CHAR;
          }
          else
          {
           error =  errors.mobile = ''; 
          }

          console.log('regex value',(/^-?[\d.]+(?:e-?\d+)?$/.test(value)));
          break;

         case 'password':        
         if(/\s/.test(value))
         {
          error = errors.password = AppConstants.VALIDATION_PASSWORD_WITH_SPACE;
         }
         else if(value.length < 8)
         {
          error = errors.password = AppConstants.VALIDATION_PASSWORD_MIN_CHAR; 
         }
         else if(value.length > 30)
         {
          error =  errors.password = AppConstants.VALIDATION_PASSWORD_MAX_CHAR;
         }
         else
         {
          error =  errors.password = ''; 
         }
         break;
       default:
         break;
     }
 
     setErrors({[name]: error});
  };

  const handleSubmit = (evt) => {
    evt.preventDefault();
    console.log(userInfo);
    fnRegister(userInfo);
    setuserInfo({ name: "", email: "", mobile: "", password: "" });
    setsuccess(true);
  };
  if (success) {
    console.log(success);
    setsuccess(false);
    return <Redirect to={"/"} />;
  }
  
  return (
    <div>
      <section className="login_container">
        <div className="container-fluid h-100">
          <div className="row h-100">
            <Col sm="5" className="reg-container">
              <div>
                <h1>Register</h1>
                <Form onSubmit={handleSubmit}>
                  <FormGroup>
                    <Label for="name">Name</Label>
                    <Input
                      type="text"
                      value={userInfo.name}
                      name="name"
                      id="name"
                      placeholder="Name"
                      required
                      onChange={(event) =>
                        {
                        setuserInfo({
                          name: event.target.value,
                          password: userInfo.password,
                          email: userInfo.email,
                          mobile: userInfo.mobile,
                        })
                        fnonChange(event)
                        }
                      }
                    />
                    {(errors.name)? <div className='error'>{errors.name}</div> : ''}
                  </FormGroup>
                  <FormGroup>
                    <Label for="email">Email</Label>
                    <Input
                      type="text"
                      value={userInfo.email}
                      name="email"
                      id="email"
                      placeholder="Email"
                      required
                      onChange={(event) =>
                            {
                            setuserInfo({
                              name: userInfo.name,
                              password: userInfo.password,
                              email: event.target.value,
                              mobile: userInfo.mobile,
                            })
                            fnonChange(event)
                          }
                      }
                    />
                    {(errors.email)? <div className='error'>{errors.email}</div> : ''}
                  </FormGroup>
                  <FormGroup>
                    <Label for="mobile">Mobile</Label>
                    <Input
                      type="text"
                      value={userInfo.mobile}
                      name="mobile"
                      id="mobile"
                      placeholder="Mobile"
                      required
                      onChange={(event) =>
                      {
                        setuserInfo({
                          name: userInfo.name,
                          password: userInfo.password,
                          email: userInfo.email,
                          mobile: event.target.value,
                        })
                        fnonChange(event)
                      }
                      }
                    />
                    {(errors.mobile)? <div className='error'>{errors.mobile}</div> : ''}
                  </FormGroup>
                  <FormGroup>
                    <Label for="password">Password</Label>
                    <Input
                      type="password"
                      value={userInfo.password}
                      name="password"
                      id="password"
                      placeholder="********"
                      required
                      onChange={(event) =>
                        {
                          setuserInfo({
                            name: userInfo.name,
                            password: event.target.value,
                            email: userInfo.email,
                            mobile: userInfo.mobile,
                          })
                          fnonChange(event)
                        }
                      }
                    />
                    {(errors.password)? <div className='error'>{errors.password}</div> : ''}
                  </FormGroup>
                  <FormGroup>
                    <Button type="submit" className="reg_btn">
                      REGISTER
                    </Button>
                    <p>
                      Having account ! <Link to="/"> Login </Link>
                    </p>
                  </FormGroup>
                </Form>
              </div>
            </Col>
          </div>
        </div>
      </section>
    </div>
  );
}

export default Register;
