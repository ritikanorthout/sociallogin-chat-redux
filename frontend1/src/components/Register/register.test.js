import React from "react";
import ReactDOM from "react-dom";
import Register from "./register";
import { Provider } from "react-redux";
import store from "../store";

it("NotFound renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <Provider store={store}>
      <Register />
    </Provider>,
    div
  );
  expect(div).toMatchSnapshot();
});
