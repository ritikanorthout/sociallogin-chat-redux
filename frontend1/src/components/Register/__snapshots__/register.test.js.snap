// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound renders without crashing 1`] = `
<div>
  <div>
     
    <section
      class="login_container"
    >
      <div
        class="container-fluid h-100"
      >
        <div
          class="row h-100"
        >
          <div
            class="reg-container col-sm-5"
          >
            <div>
              <h1>
                Register
              </h1>
              <form
                class=""
              >
                <div
                  class="form-group"
                >
                  <label
                    class=""
                    for="name"
                  >
                    Username
                  </label>
                  <input
                    class="form-control"
                    id="name"
                    name="name"
                    placeholder="Username"
                    required=""
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="form-group"
                >
                  <label
                    class=""
                    for="email"
                  >
                    Email
                  </label>
                  <input
                    class="form-control"
                    id="email"
                    name="email"
                    placeholder="Email"
                    required=""
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="form-group"
                >
                  <label
                    class=""
                    for="mobile"
                  >
                    Mobile
                  </label>
                  <input
                    class="form-control"
                    id="mobile"
                    name="mobile"
                    placeholder="Mobile"
                    required=""
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="form-group"
                >
                  <label
                    class=""
                    for="examplePassword"
                  >
                    Password
                  </label>
                  <input
                    class="form-control"
                    id="examplePassword"
                    name="password"
                    placeholder="********"
                    required=""
                    type="password"
                    value=""
                  />
                </div>
                <div
                  class="form-group"
                >
                  <button
                    class="reg_btn btn btn-secondary"
                    type="submit"
                  >
                    REGISTER
                  </button>
                  <p>
                    Having account !
                     
                    <a
                      href="/"
                    >
                       Login 
                    </a>
                     
                  </p>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;
